### PR TITLE
Add acount edit form and password change

### PR DESCRIPTION
**Pull request by txels**
Additions:
- We now expose the profile form in the account menu
- Add a view to change one's password
- Add some tests
- Automate the creation of a test user in LDAP

TODO:
- [x] Add postgres, openLDAP and creation of test user(s) to CI/CD pipeline

image

image

### CODE REVIEW

File: humans/templates/shipanaro/base.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {% extends 'shipanaro/base.html' %}
 {% load i18n static %}
 {% block extrastyle %}
-<link
+<!--link
   href="https://fonts.googleapis.com/css?family=Open+Sans"
   rel="stylesheet"
-/>
+/-->
```
Comment (txels) [inline humans/templates/shipanaro/base.html:7]: Arguable, but this speeds up page loads and I don't much see the benefit of this font :sweat_smile:

File: shipanaro/models.py [hunk 2]
```diff
@@ -6,25 +6,30 @@
...
-    (2, _('Female')),
-    (9, _('Gender non-conforming')),
+    (1, _("Male")),
+    (2, _("Female")),
+    (3, _("Agender")),
+    (4, _("Fluid")),
+    (5, _("Non-binary")),
+    (6, _("Pangender")),
+    (7, _("Transgender")),
+    (9, _("Gender non-conforming")),
```
Comment (txels) [inline shipanaro/models.py:25]: Extended to match current applicant form in https://participa.pirates.cat/form/inscripcio

File: shipanaro/templates/registration/profile.html [hunk 3]
```diff
@@ -14,8 +14,8 @@
 			{% endbuttons %}
 		</form>
 	</div>
-	<div id="profile-actions" class="col-lg-3 mb-4 pt-3 bg-light">
-		<button type="submit" class="btn btn-primary">{% trans 'Cancel membership' %}</button>
-	</div>
+	<!--div id="profile-actions" class="col-lg-3 mb-4 pt-3 bg-light">
+		<button type="submit" class="btn btn-danger">{% trans 'Cancel membership' %}</button>
+	</div-->
```
Comment (txels) [inline shipanaro/templates/registration/profile.html:19]: This button is not really working ATM so I've commented it out.

File: shipanaro/models.py [hunk 4]
```diff
@@ -47,33 +53,35 @@ class Membership(models.Model):
     contact_id = models.CharField(max_length=9)
     date_left = models.DateField(null=True, blank=True)
     drop_out = models.BooleanField(default=False)
+    # membership applications pending acceptance
+    date_accepted = models.DateField(null=True, blank=True)
+    activated = models.BooleanField(default=False)
```
Comment (txels) [inline shipanaro/models.py:58]: Once new affiliates can apply via here, we can use these fields to manage "application approval"

File: shipanaro/models.py [hunk 5]
```diff
@@ -47,33 +53,35 @@ class Membership(models.Model):
...
 
     def save(self, *args, **kwargs):
         super(Membership, self).save(*args, **kwargs)
         if self.drop_out:
-            subs = Subscription.objects.filter(member=self)
-            for sub in subs:
-                sub.delete()
-        else:
+            Subscription.objects.filter(member=self).delete()
```
Comment (txels) [inline shipanaro/models.py:63]: This results in a single DB query.

**Comment by txels** [comment]
Build is now green, with streamlined test user creation and postgres and ldap running in CI.

**Review by darccio — APPROVED** [review]

**Pull request merged**